src/main/kotlin/no/elhub/auth/features/documents/create/CreateDocumentCommand.kt

```diff
@@ -1,10 +1,34 @@
 package no.elhub.auth.features.documents.create
 
 import no.elhub.auth.features.documents.AuthorizationDocument
+import java.time.LocalDateTime
+import java.util.UUID
 
-data class CreateDocumentCommand(
-    val type: AuthorizationDocument.Type,
-    val requestedBy: String,
-    val requestedTo: String,
-    val meteringPoint: String
-)
+sealed interface CreateDocumentCommand {
+    data class ChangeOfSupplier(
+        val requestedFrom: String,
+        val requestedFromName: String,
+        val requestedBy: String,
+        val balanceSupplierContractName: String,
+        val meteringPointId: String,
+        val meteringPointAddress: String,
+
+    ) : CreateDocumentCommand
+}
+
+fun CreateDocumentCommand.toAuthorizationDocument(
```

Comment:
:art: Don't think this really should be a mapping, this responsibility belongs to the handler. The instructions we get in (command) and the entity we create is not exactly a 1-to-1 mapping. Things like validation, how we generate IDs, supplementary data which need to be fetched/generated will impact how we ultimately build the entity.